index.js

```diff
@@ -62,15 +61,41 @@ const useNavigate = (options) => {
  * Part 2, Low Carb Router API: Router, Route, Link, Switch
  */
 
-export const Router = (props) => {
-  // this little trick allows to avoid having unnecessary
-  // calls to potentially expensive `buildRouter` method.
+export const Router = ({
+  hook,
+  matcher,
+  base = "",
+  nested = false,
+  children,
+}) => {
+  const parent = useRouter();
+
+  // nested `<Router />` has the scope of its closest parent router (base path is prepended)
+  // Routers are not nested by default, but this might change in future versions
+  const proto = nested ? parent : defaultRouter;
+
+  // updates the current router with the props passed down to the component
+  const updateRouter = (router) => {
+    router.hook = hook || proto.hook;
+    router.matcher = matcher || proto.matcher;
+    router.base = proto.base + base;
+
+    // parent router reference, `undefined` when router is top-level
+    router.parent = proto === defaultRouter ? proto : undefined;
```

Comment:
@HansBrende FYI the way parent is assigned.

Reply:
Whoops, it should be other way around!

Reply:
@molefrog yes, flipped 😂 . Also please see my comment here: https://github.com/molefrog/wouter/pull/265#discussion_r1023228882 where I flesh out your requested example. Hopefully that makes sense?

Reply:
Done in f5fd814cc99ba9bb7798f8dc12ec3ebe51db32da, also added some assertions.